aplib.net-demo/Assets/Testing/AplibTests/SolveMazeTest.cs

```diff
@@ -0,0 +1,196 @@
+                bool isWallHitToRight = isHitToRight && hitRight.point.y > playerPosition.y + Mathf.Epsilon;
+                bool isWallInFront = isWallHitInFront || isWallHitToRight;
+
+                Debug.Log($"IsHoleInFront: {isHoleInFront}, IsWallInFront: {isWallInFront}");
+
+                // Return whether there is any obstacle in front of the player.
+                return isHoleInFront || isWallInFront;
+            }
+        );
+
+    /// <summary>
+    /// Whether or not an obstacle is to the right of the player, preventing it to turn right.
+    /// </summary>
+    public Belief<Transform, bool> IsObstacleToRight = new
+        (
+            reference: GameObject.Find("Player").transform,
+            getObservationFromReference: playerTransform =>
+            {
+                // Determine the eye position of the player, since the player transform is at the feet of the player.
+                // Also determine the direction that the player should look in.
+                Vector3 playerPosition = playerTransform.position;
+                Vector3 eyePosition = playerPosition + new Vector3(0, 1.75f, 0);
+                Vector3 lookFrontDirection = playerPosition - eyePosition + (playerTransform.forward + playerTransform.right);
+                Vector3 lookRightDirection = playerPosition - eyePosition + playerTransform.right;
+
+                // Shoot two raycasts from the eye of the player.
+                bool isHitInFront = Physics.Raycast(eyePosition, lookFrontDirection, out RaycastHit hitFront, 4f);
+                bool isHitToRight = Physics.Raycast(eyePosition, lookRightDirection, out RaycastHit hitRight, 4f);
+
+                Debug.DrawRay(eyePosition, lookFrontDirection, Color.blue);
+                Debug.DrawRay(eyePosition, lookRightDirection, Color.blue);
+
+                // Check if there is a hole to the right of the player.
+                bool isHoleToRight = !isHitInFront || !isHitToRight;
+
+                // Check if the raycast actually hit a wall or just the floor.
+                bool isWallHitInFront = isHitInFront && hitFront.point.y > playerPosition.y + Mathf.Epsilon;
+                bool isWallHitToRight = isHitToRight && hitRight.point.y > playerPosition.y + Mathf.Epsilon;
+                bool isWallToRight = isWallHitInFront || isWallHitToRight;
+
+                // Return whether there is any obstacle to the right of the player.
+                return isHoleToRight || isWallToRight;
+            }
+        );
+
+    /// <summary>
+    /// The name of the current room that the player is in.
+    /// </summary>
+    public Belief<Transform, string> CurrentRoom = new
+        (
+            reference: GameObject.Find("Player").transform,
+            getObservationFromReference: playerTransform =>
+            {
+                bool isHit = Physics.Raycast(playerTransform.position - new Vector3(0, Mathf.Epsilon, 0), -playerTransform.up, out RaycastHit hitRight, 4f);
+
```

Comment:
Please explain that you determine in what room you are, by testing the floor objects' name.